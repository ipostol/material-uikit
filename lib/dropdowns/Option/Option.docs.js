'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkupComponents = require('react-markup-components');

var _Option = require('./Option');

var _Option2 = _interopRequireDefault(_Option);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function () {
  return _react2.default.createElement(
    _reactMarkupComponents.Skin,
    { style: { backgroundColor: '#ffffff' } },
    _react2.default.createElement(
      'div',
      { style: { padding: '10px' } },
      _react2.default.createElement(
        _Option2.default,
        { disabled: true },
        'Test Label disabled'
      ),
      _react2.default.createElement(
        _Option2.default,
        null,
        'Test Label'
      )
    )
  );
};