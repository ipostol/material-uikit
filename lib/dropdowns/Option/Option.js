'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Option = require('./Option.scss');

var _Option2 = _interopRequireDefault(_Option);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Option = function Option(_ref) {
  var Element = _ref.element,
      children = _ref.children,
      className = _ref.className,
      size = _ref.size,
      icon = _ref.icon,
      iconClass = _ref.iconClass,
      disabled = _ref.disabled,
      rest = _objectWithoutProperties(_ref, ['element', 'children', 'className', 'size', 'icon', 'iconClass', 'disabled']);

  return _react2.default.createElement(
    Element,
    _extends({
      className: (0, _classnames2.default)(_Option2.default.general, className, _Option2.default[size], _defineProperty({}, _Option2.default.disabled, disabled))
    }, rest),
    icon && _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)(_Option2.default.icon, iconClass) },
      icon
    ),
    icon ? _react2.default.createElement(
      'span',
      null,
      children
    ) : children
  );
};

Option.propTypes = {
  children: _propTypes2.default.node,
  element: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]),
  icon: _propTypes2.default.node,
  iconClass: _propTypes2.default.string,
  className: _propTypes2.default.string,
  disabled: _propTypes2.default.bool,
  size: _propTypes2.default.string // s m l xl etc... for components height
};

Option.defaultProps = {
  size: 's',
  element: 'div'
};

exports.default = Option;