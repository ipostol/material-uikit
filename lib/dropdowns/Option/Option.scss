@import '~bidstack-theme/theme';

.general {
  line-height: 1;
  font-size: $sizeBodyS;
  color: $fontColorMain;
  display: flex;
  align-items: center;
  height: 36px;
  background-color: $white3;
  cursor: pointer;
  padding: 0 16px;
  user-select: none;
  text-decoration: none;

  &:hover {
    background-color: $white1;
  }
}

.icon {
  padding-right: 16px;
  width: 24px;
  height: 24px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.s {
  height: 36px;
}

.l {
  height: 54px;
}

.disabled {
  opacity: 0.5;
  pointer-events: none;
}
