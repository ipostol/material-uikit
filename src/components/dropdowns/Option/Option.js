import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './Option.scss';

const Option = ({
  element: Element,
  children,
  className,
  size,
  icon,
  iconClass,
  disabled,
  ...rest
}) => (
  <Element
    className={cn(styles.general, className, styles[size], { [styles.disabled]: disabled })}
    {...rest}
  >
    {icon && <div className={cn(styles.icon, iconClass)}>{icon}</div>}
    {icon ? <span>{children}</span> : children}
  </Element>
);

Option.propTypes = {
  children: PropTypes.node,
  element: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
  ]),
  icon: PropTypes.node,
  iconClass: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  size: PropTypes.string, // s m l xl etc... for components height
};

Option.defaultProps = {
  size: 's',
  element: 'div',
};

export default Option;
