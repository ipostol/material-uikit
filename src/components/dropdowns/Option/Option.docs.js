import React from 'react';
import { Skin } from 'react-markup-components';
import Option from './Option';

export default () => (
  <Skin style={{ backgroundColor: '#ffffff' }}>
    <div style={{ padding: '10px' }}>
      <Option disabled>Test Label disabled</Option>
      <Option>Test Label</Option>
    </div>
  </Skin>
);
