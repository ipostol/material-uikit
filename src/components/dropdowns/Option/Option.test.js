import React from 'react';
import { shallow } from 'enzyme';
import Option from './Option';

describe('Option component', () => {

  it('should render correctly', () => {

    const props = {};

    const wrapper = shallow(
      <Option {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('should render correctly icon props', () => {

    const props = {
      icon: <div />,
    };

    const wrapper = shallow(
      <Option {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

});
