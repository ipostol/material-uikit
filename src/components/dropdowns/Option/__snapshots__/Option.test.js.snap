// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option component should render correctly 1`] = `
<div
  className="general s"
/>
`;

exports[`Option component should render correctly icon props 1`] = `
<div
  className="general s"
>
  <div
    className="icon"
  >
    <div />
  </div>
  <span />
</div>
`;
